react/components/BuyButton/index.js

```diff
@@ -182,33 +189,57 @@ export const BuyButton = ({
     }, 500)
   }
 
-  return (
-    <Fragment>
-      {!skuItems ? (
-        <ContentLoader />
-      ) : (
-        <Button
-          block={large}
-          disabled={
-            disabled ||
-            !available ||
-            (orderFormContext && orderFormContext.loading)
-          }
-          isLoading={isAddingToCart}
-          onClick={e => shouldAddToCart && handleAddToCart(e)}
-        >
-          {available ? (
-            children
-          ) : (
-            <FormattedMessage id="store/buyButton-label-unavailable">
-              {message => (
-                <span className={handles.buyButtonText}>{message}</span>
-              )}
-            </FormattedMessage>
+  const handleClick = e => {
+    dispatch({ type: 'SET_BUY_BUTTON_CLICKED', args: { clicked: true } })
+    if (skuSelector.areAllVariationsSelected && shouldAddToCart) {
+      handleAddToCart(e)
+    }
+  }
+
+  if (!skuItems) {
+    return <ContentLoader />
+  }
+
+  const disabled = disabledProp || !available || (orderFormContext && orderFormContext.loading)
+  let children = childrenProp
+  if (!available) {
+    children = (
+      <FormattedMessage id="store/buyButton-label-unavailable">
+          {message => (
+            <span className={handles.buyButtonText}>{message}</span>
           )}
-        </Button>
+      </FormattedMessage>
+    )
+  }
+
```

Comment:
I understand what you're doing here, but I think this might be too confusing since you're kind of overriding the name of a default variable in React `children` in this case. When you use this later on in your render, you use `children`, but you're not always rendering the *actual* `children`, you see?
I would just put this conditional logic in the render itself (could even be on line 230), or create a variable to store this content, something like `const unavailableLabel = ...`, and then use it on line 230 along with the conditional check, something like:
```
{!available ? unavailableLabel : children}
```

Reply:
Oh! Then you would not even need that rename from `children` to `childrenProp`